proposals/0200-raw-string-escaping-revision-1.md

```diff
@@ -0,0 +1,343 @@
+
+The first iteration of [SE-0200](https://github.com/apple/swift-evolution/blob/master/proposals/0200-raw-string-escaping.md) proposed adopting Python's model, using `r"...raw string..."`. The proposal was returned for revision with the [following feedback](https://forums.swift.org/t/returned-for-revision-se-0200-raw-mode-string-literals/11630):
+
+> The review of SE-0200: "Raw" mode string literals 71 ran from March 16…26, 2018. The proposed is returned for revision, and should be further discussed as a pitch to coalesce further before coming up for review again.
+>
+> 
+> During the review discussion, a few issues surfaced with the proposal, including:
+>
+> The proposed r"..." syntax didn’t fit well with the rest of the language. The most-often-discussed replacement was #raw("..."), but the Core Team felt more discussion (as a pitch) is necessary.
+>
+> The proposal itself leans heavily on regular expressions as a use case for raw string literals. Several reviewers remarked that the motivation wasn’t strong enough to justify the introduction of new syntax in the language, so a revised proposal will need additional motivating examples in other domains.
+
+To move forward, the new raw string design must provide a suitable Swift-appropriate syntax that works within the language's culture and conventions.
+
+## Prior Art
+
+The following links explore the existing art in other languages. We were inspired by the [Rust raw string RFC discussion](https://github.com/rust-lang/rust/issues/9411) when researching these features.
+
+| Syntax | Language(s) | Possible in Swift? | Swifty? |
+| ----- | --------- | ------------- | -------- |
+| `'Hello, world!'` | Bourne shell, Perl, PHP, Ruby, Windows PowerShell | Yes | Yes if Rust-style multiplicity allows incorporating `'` into raw strings. May be too narrow a use-case to burn `'`. |
+| `q(Hello, world!)` | [Perl](https://en.wikipedia.org/wiki/String_literal) (alternate) | Maybe (depends on delimiter) | No |
+| `%q(Hello, world!)` | Ruby (alternate) | No (`%` is a valid prefix operator) | No |
+| `@"Hello, world!"` | [C#](https://msdn.microsoft.com/en-us/library/69ze775t.aspx), F# | Yes (but would be awful for Obj-C switchers) | No |
+| `R"(Hello, world!)"` | [C++11](https://en.cppreference.com/w/cpp/language/string_literal) | Yes | No |
+| `r"Hello, world!"` | [D](https://tour.dlang.org/tour/en/basics/alias-strings), [Python](http://wiki.c2.com/?RawStrings) | Yes | No |
+| `r#"Hello, world!"#` | [Rust](https://doc.rust-lang.org/reference/tokens.html#raw-string-literals) | Yes | Would need to drop the opening `r` and maybe change the delimiter from `#`. |
+| `raw"Hello, world!"` | Scala | Yes | No |
```

Comment:
For reference (no strong opinions, but figured may be useful to know): there are two ways in Scala, the "raw" one rarely being used in real life. 


The second one at the same time being multi-line strings (and also the most of the time used one: also for in-line strings which need the raw representation); Examples:

```
scala> raw"hello \' world" // rarely used
res0: String = hello \' world 

scala> """hello \' world""" // most often used; same syntax as the multi-line string syntax
res1: String = hello \' world

scala> "hello \' world" // normal string for reference
res2: String = hello ' world
```

I suppose this would be "Swifty" however as `"""` already exists it could cause confusion if its semantics changes.
In any case, thought that may be interesting to share.

Reply:
I will update accordingly